lib/document/enrichContractWithBaseDocument.js

```diff
@@ -18,16 +19,19 @@ function enrichContractWithBaseDocument(contract) {
   Object.keys(clonedDocuments).forEach((type) => {
     const clonedDocument = clonedDocuments[type];
 
-    const { properties: baseDocumentProperties } = documentBaseSchema;
+    const { properties: baseDocumentProperties, required: baseRequired } = documentBaseSchema;
 
     if (!clonedDocument.required) {
       clonedDocument.required = [];
     }
 
-    Object.keys(baseDocumentProperties).forEach((name) => {
-      clonedDocument.properties[name] = baseDocumentProperties[name];
-      clonedDocument.required.push(name);
-    });
+    Object.keys(baseDocumentProperties)
+      .filter(name => excludeBaseProperty.indexOf(name) === -1)
+      .forEach((name) => {
+        clonedDocument.properties[name] = baseDocumentProperties[name];
+      });
+
+    baseRequired.forEach(name => clonedDocument.required.push(name));
```

Comment:
If `required` is not defined in clonedDocument?

Reply:
This logic is happening a little bit earlier.
```
if (!clonedDocument.required) {
  clonedDocument.required = [];
}
```